include/rmm/device_vector.hpp

```diff
@@ -20,6 +20,8 @@
 
 #include <thrust/device_vector.h>
 
+#include <rmm/detail/thrust_namespace.h>
```

Comment:
So this doesn't need to go above the thrust includes?

Reply:
Currently it evaluates to a `using namespace` call which requires that namespace to exist before it is used.
What we could do is construct that namespace ( rapids::thrust ) before the using to allow this include to occur before any thrust includes

Example: https://godbolt.org/z/cE5bTTh1M

Reply:
I have updated the PR with the above technique as it removes the issue that the order of includes could break compilation.